Separate character max age into max generated and max input

LeDrascol opened the pull request:
## About The Pull Request
Adds a method to set maximum character age _input_ separately from _randomly generated_ maximum character age. When using the character age setting dialog, and when saving that value, the code will reference AGE_MAX_INPUT instead of AGE_MAX. By default; these two definitions are identical.

Originally requested as a method to remove maximum age for S.P.L.U.R.T. [in this PR](https://github.com/SPLURT-Station/S.P.L.U.R.T-Station-13/pull/618), but deemed too difficult to introduce as modular.

## Why It's Good For The Game
Allows raising or unrestricting the maximum character age a user can set _without_ causing age to randomly generate in the new range.

## Changelog
:cl:
code: Added definition for maximum character age input
/:cl:

LeDrascol committed:
Remove maximum age

Allows characters to be any age above the minimum. Does not remove or change AGE_MAX definition.

Vizoplez commented on code/modules/client/preferences_savefile.dm:
```diff
@@ -905,7 +905,7 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	socks = sanitize_inlist(socks, GLOB.socks_list)
 	socks_color = sanitize_hexcolor(socks_color, 6, FALSE, initial(socks_color))
-	age = sanitize_integer(age, AGE_MIN, AGE_MAX, initial(age))
+	age = sanitize_integer(age, AGE_MIN, INFINITY, initial(age))
```
Time to overflow the age boundary!

LeDrascol replied:
Reducing the value from INFINITY to SHORT_REAL_LIMIT should resolve this issue.

LeDrascol committed:
Reduce max age from INFINITY to SHORT_REAL_LIMIT

This effectively reduces the cap to 16777216.

LeDrascol committed:
Separate max age from max age input

The maximum character age a user can manually input is now defined by AGE_MAX_INPUT instead of AGE_MAX. These values are identical by default.

Also reverts the change to dialog box text.
